react/week4/src/GithubList.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react'
+
+
+function fetchGithub(query) {
+    const URL = `https://api.github.com/search/users?q=${query}`
+    return fetch(URL).then((response) => response.json())
+}
+
+export default class GithubList extends React.Component {
+    state = {
+        users: [], 
+        isLoading: true, 
+        searchUser: '',
+        errorMessage: null
+    }
+
+
+handleSubmit = (e) => {
+    const searchTerm = e.target.value;
+    console.log("Handle submit", searchTerm);   
+    this.setState({ 
+        searchUser: searchTerm });
+        if(searchTerm.length === 0) {
+            this.setState({
+                users: [], 
+                isLoading: false,
+            }) 
+            return
+        } 
+ fetchGithub(searchTerm).then((data) => {
+     console.log("fetch data", data, searchTerm);    
```

Comment:
Generally: avoid adding `console.log` in the commits: use them to debug, but after you are done, take them out :)

Why? when there is console.logs in 10 parts of the code and you are now coding the 11th (and adding console.log to it), the console will have a lot of messages from the other 10 parts, which will make it difficult to debug the 11th component.